### DISCUSSION TITLE
`WSL` remote extension code wants to open a directory to the Windows system, what should I do?

### DISCUSSION THREAD

**Discussion by mozhuanzuojing** [microsoft/vscode-discussions#692] [Extension Development QnA] [answered]
My ext run in a WSL environment, but I want to open a directory of Linux into `Windows`, similar to `Reveal in Explorer`. But I call `vscode.commands.executeCommand ("revealFileInOS", myFolder)` in my code, and it doesn't work.

**Comment by burkeholland** [comment, marked as answer]
Can you call "reveal in explorer" directly? Full Disclusure - Bing recommended this when I fed it the command docs article.

```
vscode.commands.executeCommand("remote-wsl.revealInExplorer", vscode.Uri.file(myFolder));
```

**Reply by mozhuanzuojing** [reply]
Thank you very much for your reply. I'll try it right away.

**Reply by Splines** [reply]
In my case, I wanted it to be flexible enough to run on wsl remotes as well as other machines. Just check for `vscode.env.remoteName` as I've done [here](https://github.com/microsoft/vscode/issues/232522#issuecomment-2446860199).
